flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/hive/HivePropertiesConverter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.flink.connector.hive;
+
+import static com.datastrato.gravitino.catalog.hive.HiveCatalogPropertiesMeta.METASTORE_URIS;
+
+import com.datastrato.gravitino.flink.connector.PropertiesConverter;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import java.util.Map;
+import java.util.Set;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.catalog.CommonCatalogOptions;
+import org.apache.flink.table.catalog.hive.HiveCatalog;
+import org.apache.flink.table.catalog.hive.factories.HiveCatalogFactoryOptions;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+public class HivePropertiesConverter implements PropertiesConverter {
+
+  private HivePropertiesConverter() {}
+
+  public static final HivePropertiesConverter INSTANCE = new HivePropertiesConverter();
+
+  public static final Map<String, String> HIVE_CONFIG_TO_GRAVITINO =
+      ImmutableMap.of(HiveConf.ConfVars.METASTOREURIS.varname, METASTORE_URIS);
+
+  private static final Set<String> SUPPORTED_OPTIONS =
+      ImmutableSet.of(
+          HiveCatalogFactoryOptions.DEFAULT_DATABASE.key(),
+          HiveCatalogFactoryOptions.HIVE_CONF_DIR.key(),
+          HiveCatalogFactoryOptions.HADOOP_CONF_DIR.key(),
+          HiveCatalogFactoryOptions.HIVE_VERSION.key(),
+          FactoryUtil.PROPERTY_VERSION.key(),
+          HiveCatalogFactoryOptions.HADOOP_CONF_DIR.key(),
+          CommonCatalogOptions.CATALOG_TYPE.key());
+
+  @Override
+  public Map<String, String> toGravitinoCatalogProperties(Configuration flinkConf) {
+    Map<String, String> gravitinoProperties = Maps.newHashMap();
+
+    String hiveConfDir = flinkConf.get(HiveCatalogFactoryOptions.HIVE_CONF_DIR);
+    String hadoopConfDir = flinkConf.get(HiveCatalogFactoryOptions.HADOOP_CONF_DIR);
+    HiveConf hiveConf = HiveCatalog.createHiveConf(hiveConfDir, hadoopConfDir);
+    hiveConf.forEach(
+        entry -> {
+          String hiveKey = HIVE_CONFIG_TO_GRAVITINO.get(entry.getKey());
+          if (hiveKey != null) {
+            gravitinoProperties.put(hiveKey, entry.getValue());
+          } else {
+            gravitinoProperties.put(entry.getKey(), entry.getValue());
```

Comment:
If a hive or hadoop configuration key not in `SUPPORTED_OPTIONS`, it stores to Gravitino, but when load properties from Gravitino, we just retrieve the kye in `SUPPORTED_OPTIONS`, is it expected?

Reply:
@FANNG1 The HiveCatalogFactory in Flink only receives `hive-conf-dir` and `hadoop-conf-dir`. The hive or Hadoop configuration should be written to the configuration files pointed by `hive-conf-dir`. So we can record the `hive-conf-dir` or `hadoop-conf-dir` in Gravitino.  So it's different with the Spark and Trino. But I want to change it now.

I use the `hive-conf-dir` and `hadoop-conf-dir` because I use the `HiveCatalogFactory.createCatalog()` to create the HiveCatalog of Flink. See https://github.com/datastrato/gravitino/pull/2635/files#diff-0a77fb25e843f0c8aa870098d39edaa050e458ee51a97c323cc858c246f3e4d9R28. The HiveCatalogFactory will validate the required options. So I have to remove some options that were not supported.

After thinking for a while, we can construct the HiveCatalog directly. So we can get all options from the Gravitino, construct the HiveConf, and pass them to the HiveCatalog.